### PR TITLE
 Add example smoke tests.

### PR DESCRIPTION
**Pull request by enisoc**
These are some quick scripts that perform smoke tests with the example controllers to check basic Metacontroller functionality.

### CODE REVIEW

File: examples/test.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+for test in */test.sh; do
+  echo -n "Running ${test}..."
+  if ! (cd "$(dirname "${test}")" && ./test.sh > ${logfile} 2>&1); then
+    echo "FAIL"
+    cat ${logfile}
+    echo "Test ${test} failed!"
+    exit 1
+  fi
+  echo "PASS"
```
Comment (crimsonfaith91) [inline examples/test.sh]: nit: `PASSED` or `ALL TESTS PASSED`

File: examples/bluegreen/test.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+kubectl apply -f my-bluegreen.yaml
+
+echo "Wait for nginx-blue RS to be active..."
+until [[ "$(kubectl get rs nginx-blue -o 'jsonpath={.status.readyReplicas}')" -eq 3 ]]; do sleep 1; done
+until [[ "$(kubectl get rs nginx-green -o 'jsonpath={.status.replicas}')" -eq 0 ]]; do sleep 1; done
+
+echo "Trigger a rollout..."
+kubectl patch bgd nginx --type=merge -p '{"spec":{"template":{"metadata":{"labels":{"new":"label"}}}}}'
+
+echo "Wait for nginx-green RS to be active..."
```
Comment (crimsonfaith91) [inline examples/bluegreen/test.sh]: Perhaps can add 2nd rollout to verify that the color switches back to blue.

File: examples/catset/test.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+echo "Create an object..."
+kubectl apply -f my-catset.yaml
+
+echo "Wait for 3 Pods to be Ready..."
+until [[ "$(kubectl get cs nginx-backend -o 'jsonpath={.status.readyReplicas}')" -eq 3 ]]; do sleep 1; done
+
+echo "Scale up to 4 replicas..."
+kubectl patch catset nginx-backend --type=merge -p '{"spec":{"replicas":4}}'
+
+echo "Wait for 4 Pods to be Ready..."
```
Comment (crimsonfaith91) [inline examples/catset/test.sh]: Perhaps can add a scaling down step, and verify additional pods are deleted.

**Comment by crimsonfaith91** [comment]
Left some nit comments. Everything else looks great!
/lgtm

**Comment by enisoc** [comment]
Comments addressed. Thanks for the suggestions!

**Pull request merged**